__tests__/setup/createPromiseQueue.js

```diff
@@ -0,0 +1,19 @@
+export default function createPromiseQueue() {
+  const resolves = [];
+  const stack = [];
+  const trigger = () => resolves.length && stack.length && resolves.shift()(stack.shift());
+
+  return {
+    push(value) {
+      stack.push(value);
+      trigger();
+    },
+    shift() {
+      const promise = new Promise(resolve => resolves.push(resolve));
+
+      trigger();
```

Comment:
I'm still not convinced lol

Reply:
Let's take it offline and let me know if you have a better solution.